src/librustc_resolve/diagnostics.rs

```diff
@@ -680,6 +680,7 @@ impl<'a> Resolver<'a> {
                                 Res::Def(DefKind::Ctor(..), did) => this.parent(did),
                                 _ => res.opt_def_id(),
                             };
+                            candidates.retain(|x: &ImportSuggestion|x.did !=did);
```

Comment:
Could you adjust the formatting here slightly, something like:
```suggestion
                            candidates.retain(|x: &ImportSuggestion| x.did != did);
```